compiler-core/src/type_/expression.rs

```diff
@@ -1046,6 +1046,57 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
                 }
             }
 
+            ClauseGuard::FieldAccess {
+                location,
+                label,
+                container,
+                ..
+            } => {
+                let typed_container = self.infer_clause_guard(*container.clone())?;
+
+                match typed_container.type_().as_ref() {
+                    Type::App { .. } => match typed_container.clone() {
+                        ClauseGuard::Var { location, name, .. } => {
+                            let container_expr = UntypedExpr::Var { location, name };
+                            match self.infer_field_access(
+                                container_expr,
+                                label.clone(),
+                                location,
+                                FieldAccessUsage::Other,
+                            )? {
+                                TypedExpr::RecordAccess { index, typ, .. } => {
+                                    Ok(ClauseGuard::FieldAccess {
+                                        location,
+                                        label,
+                                        index: Some(index + 1),
+                                        type_: typ,
+                                        container: Box::new(typed_container),
+                                    })
+                                }
+
+                                _ => panic!("Expected RecordAccess"),
```

Comment:
I think this will crash the compiler when it's a module being accessed.

Reply:
I tested it, it doesn't crash because the out of scope error for the module happens before this.